### PR TITLE
Implement no_case? and add it to variable_names

### LINKED ISSUE #208: Non alphanumeric variable names
We are using ... as a variable name in a macro parameter and I think it doesn't make sense that it would be stopped for not being snake case. I added a no_case check.

### PR DESCRIPTION
**Pull request by olafura**
Fixes #208

### CODE REVIEW

File: lib/credo/code/name.ex [hunk 1]
```diff
@@ -65,6 +65,10 @@ defmodule Credo.Code.Name do
     name |> String.match?(~r/^[a-z0-9\_\?\!]+$/)
   end
 
+  def no_case?(name) do
+    name |> String.match?(~r/^[^a-zA-Z0-9]*$/)
```
Comment (rrrene) [inline lib/credo/code/name.ex]: Is the `*$` intentional to match empty strings (I would have expected `+$`)?

Comment (olafura) [inline lib/credo/code/name.ex]: Fixed

**Comment by olafura** [comment]
I don't understand why the CI tests are failing seems like a bug. I can't find the code it's referring to.

**Comment by rrrene** [comment]
Not your fault. Thx for your work on this!

**Pull request merged**